chatsky/context_storages/database.py

```diff
@@ -8,181 +8,156 @@
+
+    def __init__(
+        self,
+        path: str,
+        rewrite_existing: bool = False,
+        configuration: Optional[_SUBSCRIPT_DICT] = None,
+    ):
         _, _, file_path = path.partition("://")
+        configuration = configuration if configuration is not None else dict()
         self.full_path = path
         """Full path to access the context storage, as it was provided by user."""
-        self.path = file_path
-        """`full_path` without a prefix defining db used"""
-        self._lock = threading.Lock()
-        """Threading for methods that require single thread access."""
-
-    def __getitem__(self, key: Hashable) -> Context:
-        """
-        Synchronous method for accessing stored Context.
-
-        :param key: Hashable key used to store Context instance.
-        :return: The stored context, associated with the given key.
-        """
-        return asyncio.run(self.get_item_async(key))
+        self.path = Path(file_path)
+        """`full_path` without a prefix defining db used."""
+        self.rewrite_existing = rewrite_existing
+        """Whether to rewrite existing data in the storage."""
+        self._subscripts = dict()
+        self._sync_lock = Lock()
+        for field in (self._labels_field_name, self._requests_field_name, self._responses_field_name):
+            value = configuration.get(field, self._default_subscript_value)
+            self._subscripts[field] = 0 if value == "__none__" else value
+
+    @staticmethod
+    def _synchronously_lock(condition: Callable[["DBContextStorage"], bool] = lambda _: True):
+        def setup_lock(method: Callable[..., Awaitable[Any]]):
+            @wraps(method)
+            async def lock(self: "DBContextStorage", *args, **kwargs):
+                if condition(self):
+                    async with self._sync_lock:
+                        return await method(self, *args, **kwargs)
+                else:
+                    return await method(self, *args, **kwargs)
+
+            return lock
+
+        return setup_lock
+
+    @staticmethod
+    def _verify_field_name(method: Callable[..., Awaitable[Any]]):
+        @wraps(method)
+        def verifier(self: "DBContextStorage", *args, **kwargs):
+            field_name = args[1] if len(args) >= 1 else kwargs.get("field_name", None)
+            if field_name is None:
+                raise ValueError(f"For method {method.__name__} argument 'field_name' is not found!")
+            elif field_name not in (self._labels_field_name, self._requests_field_name, self._responses_field_name):
+                raise ValueError(f"Invalid value '{field_name}' for method '{method.__name__}' argument 'field_name'!")
+            else:
+                return method(self, *args, **kwargs)
+
+        return verifier
 
     @abstractmethod
-    async def get_item_async(self, key: Hashable) -> Context:
+    async def load_main_info(self, ctx_id: str) -> Optional[Tuple[int, int, int, bytes, bytes]]:
```

Comment:
Create dataclass for main info:
It would look cleaner for the signatures and better annotation for what the values actually mean.